File: rxbinding-appcompat-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RxAppCompatCheckedTextView.java
package com.jakewharton.rxbinding.support.v7.widget;

import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;
import android.support.v7.widget.AppCompatCheckedTextView;
import rx.functions.Action1;

/**
 * Static factory methods for creating {@linkplain Action1 actions} for {@link AppCompatCheckedTextView}.
 */
public final class RxAppCompatCheckedTextView {

  /**
   * An action which sets the checked property of {@code view} with a boolean value.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   */
  @CheckResult @NonNull
  public static Action1<? super Boolean> check(@NonNull final AppCompatCheckedTextView view) {
    return new Action1<Boolean>() {
      @Override public void call(Boolean check) {
        view.setChecked(check);
      }
    };
  }

  public RxAppCompatCheckedTextView() {
    throw new AssertionError("No Instances.");
  }
}

File: rxbinding-appcompat-v7-kotlin/src/main/kotlin/com/jakewharton/rxbinding/support/v7/widget/RxAppCompatCheckedTextView.kt
package com.jakewharton.rxbinding.support.v7.widget

import android.support.v7.widget.AppCompatCheckedTextView
import rx.functions.Action1

/**
 * An action which sets the checked property of `view` with a boolean value.
 * 
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
public inline fun AppCompatCheckedTextView.check(): Action1<in Boolean> = RxAppCompatCheckedTextView.check(this)
